### PR TITLE
Feature/axios response interceptor feature

### PR DESCRIPTION
**Pull request by thiagosaife**

### CODE REVIEW

File: tests/axiosInterceptor.test.js [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    })
+    let error = ''
+    try {
+      await Storyblok.getAll('cdn/links')
+    } catch(err) {
+      error = err
+    }
+    expect(error).toEqual(new TypeError('Cannot read property \'data\' of undefined'));
+  })
+  it('should trhow a TypeError if NULL is returned', async () => {
```
Comment (ademarCardoso) [inline tests/axiosInterceptor.test.js:54]: A typo in this test title, `should trhow...` should be `should throw...`

**Review by ademarCardoso — APPROVED** [review]
Great job my friend

File: tests/axiosInterceptor.test.js (hunk 1 above)
Comment (thiagosaife) [reply tests/axiosInterceptor.test.js:54]: Thanks! Will correct now!

File: source/index.js [hunk 2]
```diff
@@ -48,6 +48,11 @@ class Storyblok {
       headers: headers,
       proxy: (config.proxy || false)
     })
+    if (config.responseInterceptor) {
+      this.client.interceptors.response.use((res) => {
+        return config.responseInterceptor(res);
```
Comment (onefriendaday) [inline source/index.js:53]: Small coding style issue. Please remove the semicolon ";" as in the rest of the project

Comment (thiagosaife) [reply source/index.js:53]: Sure. I'll do!

Comment (thiagosaife) [reply source/index.js]: Done!

**Review by emanuelgsouza — APPROVED** [review]
Great work

File: source/index.js [hunk 3]
```diff
@@ -48,6 +48,11 @@ class Storyblok {
       headers: headers,
       proxy: (config.proxy || false)
     })
+    if (config.responseInterceptor) {
+      this.client.interceptors.response.use((res) => {
+        return config.responseInterceptor(res)
+      });
```
Comment (onefriendaday) [inline source/index.js:54]: Still one semicolon left.

Comment (thiagosaife) [reply source/index.js]: Done!

**Comment by onefriendaday** [comment]
The only thing that is missing is the type definition here https://github.com/storyblok/storyblok-js-client/blob/master/types/index.d.ts#L39

**Comment by thiagosaife** [comment]
> The only thing that is missing is the type definition here https://github.com/storyblok/storyblok-js-client/blob/master/types/index.d.ts#L39

@onefriendaday I've just set the type to either the function parameter as the function return to any, because I don't think the type of the response can be controlled. In most cases, these values are arrays or objects and, in case of error, Promises. But I guess the user would like to return a string, or a number for example. Do you think this acceptable?

**Pull request merged**